ledger/participant-integration-api/src/main/scala/platform/store/packagemeta/PackageMetadataView.scala

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.packagemeta
+
+import com.daml.lf.archive.Decode
+import com.daml.lf.data.Ref
+import com.daml.daml_lf_dev.DamlLf
+import com.daml.platform.store.packagemeta.PackageMetadataView._
+import scalaz._
+
+trait PackageMetadataView {
+  def update(metadataDefinitions: MetadataDefinitions): Unit
+
+  def read(): PackageMetadata
+}
+
+object PackageMetadataView {
+  def create: PackageMetadataView = new PackageMetaDataViewImpl
+
+  trait PackageMetadata {
+    def interfaceImplementedBy(interface: Ref.Identifier): Set[Ref.Identifier]
+    def interfaceExists(interface: Ref.Identifier): Boolean
+    def templateExists(template: Ref.Identifier): Boolean
+  }
+
+  final case class MetadataDefinitions private[packagemeta] (
+      templates: Set[Ref.Identifier] = Set.empty,
+      interfaces: Set[Ref.Identifier] = Set.empty,
+      interfacesImplementedBy: Map[Ref.Identifier, Set[Ref.Identifier]] = Map.empty,
+  )
+
+  object MetadataDefinitions {
+    implicit val monoid = new Monoid[MetadataDefinitions] {
```

Comment:
besides the generative monoid law testing: is there some additional benefit to use a monoid (pull in scalaz etc) here?
if not how about moving the monoid definition next to the test, and simply adding an append in the code?
(not sure the scalaz clutter helps the readability here)

Reply:
great idea, i did not like it either that i inject scalaz for this trivial append.. will do so